app.js

```diff
@@ -22,91 +13,167 @@ const belgaConfig = {
+  }
+});
 
-app.get('/fetchTestMailCampaign/:id', async (req, res, next) => {
-  const campaign_id = req.params.id;
+app.get('/mail-campaign-content/:id', async (req, res) => {
+  const campaignId = req.params.id;
   try {
-    console.time('FETCH CAMPAIGN PREVIEW');
-    const campaignHTML = await mailchimp.get({
-      path: `/campaigns/${campaign_id}/content`
+    const mailchimpCampaign = await mailchimpService.getCampaignContent(campaignId);
+    res.send({
+      status: ok, statusCode: 200, data: {
+        'type': 'mail-chimp-campaign',
+        'id': campaignId,
+        'attributes': {
+          'html': mailchimpCampaign.html,
+        }
+      }
+    });
+  } catch (err) {
+    console.error(err);
+    res.status(500).send({
+      error: {
+        status: 500,
+        title: 'Get Mailchimp campaign failed.',
+        detail: (err.message || 'Something went wrong while fetching the mailchimp campaign.')
+      }
     });
-    console.timeEnd('FETCH CAMPAIGN PREVIEW');
-    res.send({ body: campaignHTML.html });
-  } catch (error) {
-    next(error);
   }
 });
 
-app.get('/fetchTestMailCampaignMetaData/:id', async (req, res, next) => {
-  const campaign_id = req.params.id;
+
+app.delete('/mail-campaign/:id', async (req, res) => {
+  const campaignId = req.params.id;
   try {
-    console.time('FETCH CAMPAIGN METADATA');
-    const campaignHTML = await mailchimp.get({
-      path: `/campaigns/${campaign_id}`
+    await mailchimpService.deleteCampaign(campaignId);
+    res.send({status: ok, statusCode: 200});
+  } catch (err) {
+    console.error(err);
+    res.status(500).send({
+      error: {
+        status: 500,
+        title: 'Delete mail campaign failed.',
+        detail: (err.message || 'Something went wrong while deleting mail campaign.')
+      }
     });
-    console.timeEnd('FETCH CAMPAIGN METADATA');
-    res.send({ body: campaignHTML });
-  } catch (error) {
-    next(error);
   }
 });
 
-app.delete('/deleteMailCampaign/:id', async (req, res) => {
-  const campaign_id = req.params.id;
-  if (!campaign_id) {
-    throw new Error('Request parameter campaign_id can not be null');
+app.post('/belga', async (req, res) => {
+  const meetingId = req.body.data.meetingId;
+  try {
+    await belgaService.generateXML(meetingId, true);
+    res.send({status: ok, statusCode: 200, data: {type: 'belga-campaign'}});
+  } catch (err) {
+    console.error(err);
+    res.status(500).send({
+      error: {
+        status: 500,
+        title: 'Send to Belga failed.',
+        detail: (err.message || 'Something went wrong while sending to Belga.')
+      }
+    });
   }
-  const deleted = await mailchimpService.deleteCampaign(campaign_id);
-  res.send({ deleted });
 });
 
-app.get('/xml-newsletter/:agenda_id', async (req, res) => {
-  let agendaId = req.params.agenda_id;
-  if (!agendaId) {
-    throw new Error('No agenda_id provided.');
+app.get('/belga/:meeting-id', async (req, res) => {
+  let meetingId = req.params.meeting - id;
```

Comment:
Werkt dit!?
Autoformatting kan handig zijn, maar we moeten wel blijven nadenken...